File: python/redispy/sentineltls.py
import redis
import redis.sentinel
import sys

# using workaround from https://github.com/andymccurdy/redis-py/issues/1306
class SSLSentinelManagedConnection(
    redis.sentinel.SentinelManagedConnection, redis.SSLConnection
):
    def __init__(self, **kwargs):
        self.connection_pool = kwargs.pop(
            "connection_pool"
        )  # replicate the init code from SentinelManagedConnection
        redis.SSLConnection.__init__(
            self, **kwargs
        )  # we cannot use super() here because it is not first in the MRO


def main():
    if not (len(sys.argv) == 4 or len(sys.argv) == 5):
        print("Usage: sentinelhost sentinelport service password")
        exit(1)
    sentinel_host = sys.argv[1]
    sentinel_port = sys.argv[2]
    service_name = sys.argv[3]

    password = None
    if len(sys.argv) == 5:
        password = sys.argv[4]

    # you can add more sentinels to the list
    sentinel_list = [(sentinel_host, sentinel_port)]
    s = redis.sentinel.Sentinel(
        sentinel_list,
        sentinel_kwargs={"ssl": True, "ssl_cert_reqs": None},
        **{
            "connection_class": SSLSentinelManagedConnection,
            "password": password,
            "ssl_keyfile": "../../testscripts/tls/db_key.pem",
            "ssl_certfile": "../../testscripts/tls/db_cert.pem",
            "ssl_cert_reqs": "required",
            "ssl_ca_certs": "../../testscripts/tls/ca_cert.pem",
        }
    )
    r = s.master_for(service_name, socket_timeout=3.0)

    print("Connected to Redis")
    print("Set: {}".format(r.set("foo", "bar")))
    print("Get: {}".format(r.get("foo")))

    # just for completeness close the connection pool
    r.connection_pool.disconnect()


if __name__ == "__main__":
    main()

File: cplusplus/redisplusplus/sentineltls.cpp
#include <sw/redis++/redis++.h>

#include <iostream>

using namespace sw::redis;

int main(int argc, char* argv[]) {
  if (!(argc == 4 || argc == 5)) {
    std::cout << "Usage: sentinelhost sentinelport service password "
              << std::endl;
    return 1;
  }

  auto sentinel_host = argv[1];
  auto sentinel_port = std::stoi(argv[2]);
  auto service = argv[3];

  //sentinel options. 
  SentinelOptions sentinel_opts;
  sentinel_opts.nodes = {{sentinel_host, sentinel_port}};
  sentinel_opts.connect_timeout = std::chrono::milliseconds(1000);
  sentinel_opts.socket_timeout = std::chrono::milliseconds(1000);
  auto sentinel = std::make_shared<Sentinel>(sentinel_opts);

  //connection options note timeouts are required
  ConnectionOptions opts;
  if (argc == 5) {
    opts.password = argv[4];  // password
  }
  opts.connect_timeout = std::chrono::milliseconds(1000);
  opts.socket_timeout = std::chrono::milliseconds(1000);
  opts.tls.enabled = true;
  opts.tls.cert = "../../testscripts/tls/db_cert.pem";
  opts.tls.key = "../../testscripts/tls/db_key.pem";
  opts.tls.cacert = "../../testscripts/tls/ca_cert.pem";
  ConnectionPoolOptions pool_opts;
  pool_opts.size = 3;
  pool_opts.wait_timeout = std::chrono::milliseconds(100);

  auto redis = Redis(sentinel, service, Role::MASTER, opts, pool_opts);

  auto v1 = redis.set("foo", "bar");
  if (v1) {
    std::cout << "Set: " << v1 << std::endl;
  } else {
    std::cout << "Could not set value" << std::endl;
  }

  auto v2 = redis.get("foo");
  if (v2) {
    std::cout << "Get: " << *v2 << std::endl;
  } else {
    std::cout << "Could not get value" << std::endl;
  }
  return 0;
}
